_ml-commons-plugin/model-serving-framework.md

```diff
@@ -0,0 +1,250 @@
+
+## Upload model to OpenSearch
+
+Use the URL upload operation for models that already exist on another server, such as GitHub or S3. 
+
+```
+POST /_plugins/_ml/models/_upload
+```
+
+The URL upload method requires the following request fields:
+
+Field | Data Type | Description
+:---  | :--- | :--- 
+`name`| string | The name of the model. |
+`version` | string | The version number of the model. Since OpenSearch does not enforce a specific version schema for models, you can choose any number or format that makes sense for your models. |
+`model_format` | string | The portable format of the model file. Currently only supports `TORCH_SCRIPT`. |
+`model_config` | string | The model's configuration, including the `model_type`, `embedding_dimension`, and `framework_type`. |
+`url` | string | The URL where the model is located. |
+
+#### Sample request
+
+The following sample request uploads version `1.0.0` of an NLP sentence transformation model named `all-MiniLM-L6-v2`.
+
+```json
+POST /_plugins/_ml/models/_upload
+{
+  "name": "all-MiniLM-L6-v2",
+  "version": "1.0.0",
+  "description": "test model",
+  "model_format": "TORCH_SCRIPT",
+  "model_config": {
+    "model_type": "bert",
+    "embedding_dimension": 384,
+    "framework_type": "sentence_transformers",
+  },
+  "url": "https://github.com/opensearch-project/ml-commons/raw/2.x/ml-algorithms/src/test/resources/org/opensearch/ml/engine/algorithms/text_embedding/all-MiniLM-L6-v2_torchscript_sentence-transformer.zip?raw=true"
+}
+```
+
+#### Sample response
+
+OpenSearch responds with the `task_id` and task `status`.
+
+```json
+{
+  "task_id" : "ew8I44MBhyWuIwnfvDIH", 
+  "status" : "CREATED"
+}
+```
+
+To see the status of your model upload, enter the `task_id` into the [task API]({{site.url}}{{site.baseurl}}/ml-commons-plugin/api#get-task-information).
+
+## Load the model
+
+The load model operation reads the model's chunks from the model index, then creates an instance of the model to cache into memory. The bigger the model, the more chunks the model is split. The more chunks a model index contains, the longer it takes for the model to load into memory.
+
+### Get the `model_id`
+
+To load a model, you need the `model_id`. To find the `model_id`, take the `task_id` from the model's upload operations API response and use the `GET _ml/tasks` API.
+
+This example request uses the `task_id` from the upload example.
+
+```json
+GET /_plugins/_ml/tasks/ew8I44MBhyWuIwnfvDIH
+
+OpenSearch responds with the `model_id`. 
+
+```json
+{
+  "model_id" : "WWQI44MBbzI2oUKAvNUt", 
+  "task_type" : "UPLOAD_MODEL",
+  "function_name" : "TEXT_EMBEDDING",
+  "state" : "COMPLETED",
+  "worker_node" : "KzONM8c8T4Od-NoUANQNGg",
+  "create_time" : 3455961564003,
+  "last_update_time" : 3216361373241,
+  "is_async" : true
+}
+```
+
+### Load model from index
+
+With the `model_id`, you can now load the model from the model's index, then create an instance of the model to save in an ML node's available cache.
```

Comment:
```suggestion
With the `model_id`, you can load the model from the model's index, then create an instance of the model to save in an ML node's available cache.
```

Reply:
"then create an instance of the model to save in an ML node's available cache."  This sentences seems hint user they should "create an instance of model" by themselves. Actually this load model API will do this automatically.

How about change to 
```suggestion
Once model uploaded, you can use load model API to deploy model to ML nodes. The load model API will read model chunks from the model index, then create an instance of the model and save in ML node's cache.
```